packages-exp/auth-exp/src/core/persistence/persistence_user_manager.test.ts

```diff
@@ -0,0 +1,182 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as sinon from 'sinon';
+import * as chai from 'chai';
+import { inMemoryPersistence } from './in_memory';
+import { PersistenceType, Persistence } from '.';
+import { expect } from 'chai';
+import { testUser, mockAuth } from '../../../test/mock_auth';
+import { PersistenceUserManager } from './persistence_user_manager';
+import * as sinonChai from 'sinon-chai';
+import { UserImpl } from '../user/user_impl';
+
+chai.use(sinonChai);
+
+function makePersistence(
+  type = PersistenceType.NONE
+): {
+  persistence: Persistence;
+  stub: sinon.SinonStubbedInstance<Persistence>;
+} {
```

Comment:
can we name this interface, even it is just for tests

Reply:
Discussed offline, keeping it this way